src/common/lib/python/common_extension.cc

```diff
@@ -831,13 +842,26 @@ int is_simple_value(PyObject *value, int *num_elements_contained) {
       PyTuple_Size(value) < RayConfig::instance().size_limit()) {
     for (Py_ssize_t i = 0; i < PyTuple_Size(value); ++i) {
       if (!is_simple_value(PyTuple_GetItem(value, i), num_elements_contained)) {
-        return 0;
+        return false;
       }
     }
     return (*num_elements_contained <
             RayConfig::instance().num_elements_limit());
   }
-  return 0;
+  if (PyArray_CheckExact(value)) {
+    PyArrayObject *array = reinterpret_cast<PyArrayObject *>(value);
+    if (PyArray_NBYTES(array) > RayConfig::instance().num_elements_limit()) {
```

Comment:
This looks a little confusing, why isn't this subsumed by
```
+    *num_elements_contained += PyArray_NBYTES(array);
+    return (*num_elements_contained <
+            RayConfig::instance().num_elements_limit());
```
below?

Reply:
Fixed, thx.